client/src/components/CarsFilter/index.tsx

```diff
@@ -0,0 +1,277 @@
+    mileage: 15800,
+    quality: 90,
+    isGoodPrice: true,
+    location: 'palestine - alnazreth',
+    state: 'on-market',
+    transmission: 'auto',
+    features: [
+      'Rear mirror camera',
+      'Brake assist',
+      'Reversing Camera',
+      'Collision warning system',
+      'Navigation',
+    ],
+    description: 'This car had an accedent in the left side but the quality of the body generally is good',
+    fuel: 'petrol',
+    customerId: 3,
+  },
+  {
+    brand: 'Infiniti',
+    model: 'QX70 3.7L V6',
+    price: 16500,
+    year: 2015,
+    mileage: 191000,
+    quality: 70,
+    isGoodPrice: false,
+    location: 'palestine - gaza',
+    state: 'under-check',
+    transmission: 'auto',
+    features: [
+      'Navigation',
+      'Bluetooth',
+      'Brake assist',
+      'Collision warning system',
+      'Auto dimming mirror',
+      'Headsup display',
+      'Automated Parking',
+      'Adaptive cruis control',
+    ],
+    description: '',
+    fuel: 'petrol',
+    customerId: 2,
+  },
+];
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+
+const getYears = ():Array<string> => {
+  const yearsArr = [];
+  const yearNow = new Date().getFullYear();
+
+  for (let i = 1990; i <= yearNow; i += 1) {
+    yearsArr.push(`${i}`);
+  }
+  return yearsArr;
+};
+
+function CarsFilter() {
+  const { state } = useLocation();
+
+  const [brand, setBrand] = useState(state.brand);
+  const [model, setModel] = useState('');
+  const [mileage, setMileage] = useState(100000);
+  const [year, setYear] = useState(null);
+  const [fuel, setFuel] = useState('');
+  const [maxPrice, setMaxPrice] = useState(0);
+  const [isGoodPrice, setPriceBool] = useState(false);
+
+  const changePriceType = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setPriceBool(event.target.checked);
+  };
+
+  const changeBrand = (e: any, values: SetStateAction<any>) => {
+    setBrand(values);
+  };
+  const changeModel = (e: any, values: SetStateAction<any>) => {
+    setModel(values);
+  };
+  const changeYear = (e: any, values: SetStateAction<any>) => {
+    setYear(values);
+  };
+  const changeMileage = (e: any, values: SetStateAction<any>) => {
+    setMileage(values * 10000);
+  };
+  const changefuelType = (e: React.SyntheticEvent<Element, Event>, values: SetStateAction<any>) => {
+    setFuel(values);
+  };
+  const changeMaxPrice = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const num = Number(event.target.value);
+    setMaxPrice(num);
+  };
+
+  useEffect(() => {
+    // here we will fetch the data and set it on a props that it defined in parent component
+    console.log([...cars]);
+  }, [brand, model, mileage, year, fuel, isGoodPrice]);
+
+  return (
+    <section className="filter">
+      <fieldset className="filters">
+        <p className="title">Cars filter</p>
+        <Autocomplete
+          disablePortal
+          id="combo-box"
+          onChange={changeBrand}
+          options={brands.map((e) => e.brand)}
```

Comment:
give it a default value as state.brand
```suggestion
          options={brands.map((e) => e.brand)}
           defaultValue={state?.brand || null}

```